### PR TITLE
fix multiples details

### PR DESCRIPTION
**Pull request by eliegaboriau**
This PR : 
- removes the organization from the author selection field when creating a post as a petition's author.
- It removes the author field if no user groups, to replace it with a hidden field.
- It authorizes the user to modify only his posts
- It adds warning message when no posts are published
- It adds warning message when the initiative is validating
- It removes the creation button when the initiative or the component are not published

How to test :
- Create an initiative
- Publish it
- Create post
- See you can't select organization
- Go to admin dashboard and try to create post in the same component
- See you can select organization

- Go to the posts
- Remove every post
- See the warning message
- Send the initiative to technical validation
- Go back to posts
- See the warning message and see you can't create posts

### CODE REVIEW

**Comment by paulinebessoles** [comment]
@eliegaboriau could you completely delete the "Decidim author" field from both the new and edit form? 
We'll not use the users groups feature for the CESE

**Comment by eliegaboriau** [comment]
@paulinebessoles It should be okay now

**Review by luciegrau — APPROVED** [review]
The fix works great for me, thank you for all the improvements 😁

File: app/helpers/decidim/blogs/posts_select_helper.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Blogs
+    # Custom helpers used in posts views
+    module PostsSelectHelper
+      include Decidim::ApplicationHelper
+      include SanitizeHelper
+
+      def fo_post_author_select_field(form, name, _options = {})
```
Comment (Quentinchampenois) [inline app/helpers/decidim/blogs/posts_select_helper.rb:10]: This method is a bit heavy, we should think about a refactor to simplify reading

```suggestion
            def fo_post_author_select_field(form, name, _options = {})
        select_options = [
          [current_user.name, current_user.id]
        ]
        select_options << [form.object.author.name, form.object.author.id] unless !form.object.author || select_options.pluck(1).include?(form.object.author.id)
        select_options += user_groups

        return form.select(name, select_options) if select_options.size > 1

        form.hidden_field(name, value: select_options.first[1])
      end

      private

      def user_groups
        return [] unless current_organization.user_groups_enabled?

        Decidim::UserGroups::ManageableUserGroups.for(current_user).verified.map { |user_g| [user_g.name, user_g.id] }
      end
```

File: app/overrides/decidim/blogs/posts/index/overriden_message.html.erb.deface [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!-- insert_top 'div.row.column' -->
+
+<% if paginate_posts.empty? %>
+    <div class="callout warning">
+        <%= t ".no_post_warning" %>
+    </div>
+<% end %>
+<% if current_participatory_space&.author == current_user && current_participatory_space&.state == "validating" %>
```
Comment (Quentinchampenois) [inline app/overrides/decidim/blogs/posts/index/overriden_message.html.erb.deface]: I think not every participatory spaces responds to `:author`, it occurs a crash on processes blogs

**Review by Quentinchampenois — COMMENTED** [review]
On the product part it works as expected ! 

I noted some refactoring possible, also i think we should consider adding more specs to avoid regression like 
[explore_posts_spec.rb](https://github.com/decidim/decidim/blob/develop/decidim-blogs/spec/system/explore_posts_spec.rb) because I found a internal server error on processes blogs index view

**Comment by eliegaboriau** [comment]
Thanks for your review @Quentinchampenois , It should be okay now

**Pull request merged**